src/home/celebration/NftCelebration.tsx

```diff
@@ -0,0 +1,252 @@
+import GorhomBottomSheet, { BottomSheetBackdrop, BottomSheetView } from '@gorhom/bottom-sheet'
+import { BottomSheetDefaultBackdropProps } from '@gorhom/bottom-sheet/lib/typescript/components/bottomSheetBackdrop/types'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { HomeEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { BottomSheetRefType } from 'src/components/BottomSheet'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import { nftCelebrationDisplayed } from 'src/home/actions'
+import ConfettiCelebration from 'src/home/celebration/ConfettiCelebration'
+import { lastDisplayedNftCelebration } from 'src/home/selectors'
+import ImageErrorIcon from 'src/icons/ImageErrorIcon'
+import { nftsLoadingSelector, nftsWithMetadataSelector } from 'src/nfts/selectors'
+import { NftWithMetadata } from 'src/nfts/types'
+import { getDynamicConfigParams, getFeatureGate } from 'src/statsig'
+import { DynamicConfigs } from 'src/statsig/constants'
+import { StatsigDynamicConfigs, StatsigFeatureGates } from 'src/statsig/types'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { vibrateSuccess } from 'src/styles/hapticFeedback'
+import { Spacing } from 'src/styles/styles'
+
+export default function NftCelebration() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  const insets = useSafeAreaInsets()
+  const insetsStyle = { paddingBottom: Math.max(insets.bottom, Spacing.Regular16) }
+
+  const bottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const [showAnimation, setShowAnimation] = useState(false)
```

Comment:
since the bottom sheet and confetti animation are 2 separate things being coordinated in this file, it'd be helpful if the variable names were more specific to the thing it relates to?